ledger/state.go

```diff
@@ -701,7 +701,7 @@ func (ls *LedgerState) ledgerProcessBlocks() {
 						break
 					}
 					// Enable validation using the k-slot window from ShelleyGenesis.
-					if !shouldValidate && i == 0 {
+					if i == 0 {
```

Comment:
We should keep this guard to ensure it only switches validation ON, and that we don't have to do these calculations for each block batch once validation has been enabled